classes/Form.php

```diff
@@ -402,7 +402,13 @@ public function uploadFiles()
             (array) $settings,
             ['name' => $post['name']]
         );
-
+        // Allow plugins to adapt settings for a given post name
+        // Useful if schema retrieval is not an option, e.g. dynamically created forms
+        $event = $grav->fireEvent('onFormUploadSettings', new Event(['settings' => $settings, 'post' => $post]));
+        if (isset($event->settings)) {
```

Comment:
This is not really needed as settings is an object and is passed as a reference anyway. So modifying the settings inside the event will always modify them in here, too.

The question I had was if we should allow the plugin to modify $post as well. It could be useful, but I'm not sure if it is worth it.